Java/core/src/main/java/com/amazon/randomcutforest/store/SmallIndexManager.java

```diff
@@ -75,6 +93,39 @@ public SmallIndexManager(short[] freeIndexes, short freeIndexPointer) {
         }
     }
 
+    public SmallIndexManager(int capacity, short[] freeIndexes, short freeIndexPointer) {
+        checkNotNull(freeIndexes, "freeIndexes must not be null");
+
+        this.capacity = (short) capacity;
+        this.freeIndexes = freeIndexes;
+        this.freeIndexPointer = freeIndexPointer;
+
+        occupied = new BitSet(capacity);
+        occupied.set(0, capacity);
+
+        for (int i = 0; i < freeIndexPointer; i++) {
+            if (i < freeIndexes.length) {
+                occupied.clear(freeIndexes[i]);
+            } else {
+                occupied.clear(capacity - i - 1);
+            }
+        }
+
+    }
+
+    public SmallIndexManager(int capacity, int oldCapacity) {
```

Comment:
Could we do this in a public static method that invokes `SmallIndexManager(capacity, occupied)`?

Reply:
Absolutely. Done.